internal/recover/recover.go

```diff
@@ -90,10 +307,8 @@ func ValidateMemberChanges(oldMembers []cluster.DqliteMember, newMembers []clust
 	for _, newMember := range newMembers {
 		memberValid := false
 		for _, oldMember := range oldMembers {
-			// FIXME: Allow changing member addresses as part of cluster recovery
 			membersMatch := newMember.DqliteID == oldMember.DqliteID &&
-				newMember.Name == oldMember.Name &&
-				newMember.Address == oldMember.Address
+				newMember.Name == oldMember.Name
```

Comment:
Should we do any checks here that addresses are unique, or are real addresses, before we proceed?

Also, should we ensure the new set of roles are valid? Like for example if someone sets all the roles to `spare`, we should error out.

Reply:
Both good ideas; now it checks for parseable addrs and at least one voter

Reply:
I thought that we needed a majority of nodes to be voters. Are there valid cases for most members not being voters?

Reply:
That makes me wonder, what happens when we leave the max voter count at 3, but then make all the nodes voters with the recovery command. Do they all immediately re-adjust to 3 voters?

Reply:
That's what it [looks](https://github.com/canonical/go-dqlite/blob/master/app/app.go#L600) like [to me](https://github.com/canonical/go-dqlite/blob/master/app/roles.go#L189); can test if you'd like.